third_party/blink/renderer/core/html/media/html_media_element.cc

```diff
@@ -385,6 +392,21 @@ std::ostream& operator<<(std::ostream& stream,
 // static
 MIMETypeRegistry::SupportsType HTMLMediaElement::GetSupportsType(
     const ContentType& content_type) {
+#if BUILDFLAG(IS_COBALT)
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+  // Interupt Chromium's IsTypeSupported() from here for better performance.
+  SbMediaSupportType support_type = SbMediaCanPlayMimeAndKeySystem(
+      content_type.Raw().DeprecatedLower().Ascii().c_str(), "");
+  switch (support_type) {
+    case kSbMediaSupportTypeNotSupported:
+      return MIMETypeRegistry::kNotSupported;
+    case kSbMediaSupportTypeMaybe:
+      return MIMETypeRegistry::kMaybeSupported;
+    case kSbMediaSupportTypeProbably:
+      return MIMETypeRegistry::kSupported;
+  }
+#endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
+#else   // BUILDFLAG(IS_COBALT)
```

Comment:
When BUILDFLAG(IS_COBALT) is true but BUILDFLAG(USE_STARBOARD_MEDIA) is false, there is no implementation at all.  If this is intended, add a `NOTREACHED()` and `return MIMETypeRegistry::kSupported;` there.
On the other hand, feel free to leave as is, but create a bug so we can discuss how this should be handled, as it can be confusing.

Reply:
Yes, that's intended. For now, `IS_COBALT` and `USE_STARBOARD_MEDIA`  will be true together.

Reply:
I agree that is not ideal. If turning `#else` to `#endif // BUILDFLAG(IS_COBALT)` directly, I believe we can find a way to solve problems of naming, unreachable code (I'm not sure if unreachable code is or will be treated as errors), but I don't know if it's worth it. Also @andrewsavage1. And I think if we don't explicitly have an else here, in the future when infra team refactor the macros, they will think the chromium code is needed.

Reply:
It's ok to keep this as is, and it'll be great to have input from @andrewsavage1